src/software/thunderscope/binary_context_managers/game_controller.py

```diff
@@ -21,29 +22,61 @@
 
 
 class Gamecontroller(object):
-    """ Gamecontroller Context Manager """
+    """Gamecontroller Context Manager"""
 
     CI_MODE_LAUNCH_DELAY_S = 0.3
     REFEREE_IP = "224.5.23.1"
     CI_MODE_OUTPUT_RECEIVE_BUFFER_SIZE = 9000
 
-    def __init__(self, supress_logs: bool = False) -> None:
+    def __init__(self, supress_logs: bool = False, use_conventional_port=False) -> None:
         """Run Gamecontroller
 
         :param supress_logs: Whether to suppress the logs
+        :not_launch_gc Whether to launch the gamecontroller or not!
         """
+
         self.supress_logs = supress_logs
 
-        # We need to find 2 free ports to use for the gamecontroller
-        # so that we can run multiple gamecontroller instances in parallel
-        self.referee_port = self.next_free_port()
         self.ci_port = self.next_free_port()
+        # we are not using conventional by default since most of the time
+        # we are not in competition and. Thus, we would be conflicting with
+        # each other if we are using conventional port
+        self.referee_port = self.next_free_port(random.randint(1024, 65535))
+        if use_conventional_port:
+            if not self.is_valid_port(40000):
+                raise OSError("Cannot use port 40000 for Gamecontroller")
+
+            self.referee_port = 40000
 
+        self.ci_port = self.next_free_port()
         # this allows gamecontroller to listen to override commands
         self.command_override_buffer = ThreadSafeBuffer(
             buffer_size=2, protobuf_type=ManualGCCommand
         )
 
+    @staticmethod
+    def get_referee_port_staticmethod(gamecontroller: Gamecontroller):
```

Comment:
why not make it a class function?
```suggestion
    def get_referee_port(self):
```

Reply:
oh there already is a `get_referee_port`. why do we have this function at all?

Reply:
This function is used situation like the following: 

https://github.com/Mr-Anyone/Thunderbot_Software/blob/cf668bfeaff698097aadce1d01c60b4c731322c6/src/software/thunderscope/thunderscope_main.py#L322. 

In otherwords, the instance of gamecontroller has type None, but we want a function or staticmethod to return the default port of the referee. 

Since the gamecontroller has type None, we cannot call self.get_referee_port, but Gamecontroller.get_referee_port_staticmethod instead. In other words, if the gamecontroller has type None, we want to listen to the default port.

Reply:
Hmm interesting. In that case I personally would say `Gamecontroller.get_referee_port_static` might be a more suited name.

Reply:
I have changed the name.